### PR TITLE
Status Sign Rework

### LINKED ISSUE #354: Status signs are no longer functional
**Describe the bug**
When adding a flyblock that can be comprised of multiple blocks for example [5, 125, 126] (wood, wooden slabs and wooden double slabs) only the first flyblock in the list is taken into account when calculating it on the status sign.

**To Reproduce**
1. Make craft with a flyblock that has multiple IDs.
2. Pilot that craft and update the status sign.

**Expected behavior**
All block IDs in the list should be counted into the status sign, just like in the sinking calculation.

**Versions (please complete the following information):**
 - Movecraft 7.0 beta 7
 - Java 1.8.0_271
 - Minecraft/Paper 1.12.2

### PR DESCRIPTION
**Pull request by drfiveminusmint**
### Describe in detail what your pull request accomplishes
Hopefully the final go at this one, this PR should actually fix status signs! Had to delete and reopen this because I kind of couldn't figure out how to get the old branch to be updated properly, sorry. The new implementation will cache the data, hopefully bringing some better performance, but in any case will make status signs function properly.

### Related issues:
- This will fix #354 

### Checklist
- [x] Proper internationalization
- [x] Tested

### CODE REVIEW

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/async/AsyncManager.java [hunk 1]
```diff
@@ -24,12 +24,7 @@
 import net.countercraft.movecraft.async.rotation.RotationTask;
 import net.countercraft.movecraft.async.translation.TranslationTask;
 import net.countercraft.movecraft.config.Settings;
-import net.countercraft.movecraft.craft.Craft;
-import net.countercraft.movecraft.craft.CraftManager;
-import net.countercraft.movecraft.craft.CraftStatus;
-import net.countercraft.movecraft.craft.PilotedCraft;
-import net.countercraft.movecraft.craft.PlayerCraft;
-import net.countercraft.movecraft.craft.SinkingCraft;
+import net.countercraft.movecraft.craft.*;
```
Comment (TylerS1066) [inline modules/Movecraft/src/main/java/net/countercraft/movecraft/async/AsyncManager.java:27]: Please reconfigure your IDE to not use wildcard imports and fix.

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/async/AsyncManager.java [hunk 2]
```diff
@@ -642,6 +663,12 @@ public CraftStatus checkCraftStatus(@NotNull Craft craft) {
         if (totalNonNegligibleBlocks == 0)
             isSinking = true;
 
+        if (craft instanceof BaseCraft) {
+            BaseCraft b = ((BaseCraft) craft);
+            b.updateMaterials(materials);
+            b.setTotalFuel(fuel);
+        }
+
```
Comment (TylerS1066) [inline modules/Movecraft/src/main/java/net/countercraft/movecraft/async/AsyncManager.java:671]: Like I said in the PR from ccorp, we should aim to use `BaseCraft` minimally and provide as much as we can in `Craft`.

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/craft/BaseCraft.java [hunk 3]
```diff
@@ -368,9 +369,10 @@ public int getTickCooldown() {
             return type.getIntProperty(CraftType.SINK_RATE_TICKS);
 
         if (materials.isEmpty()) {
-            for (MovecraftLocation location : hitBox) {
-                materials.add(location.toBukkit(w).getBlock().getType());
-            }
+            //for (MovecraftLocation location : hitBox) {
+            //    materials.add(location.toBukkit(w).getBlock().getType());
+            //}
+            return 20;
```
Comment (TylerS1066) [inline modules/Movecraft/src/main/java/net/countercraft/movecraft/craft/BaseCraft.java:375]: Why is this this way?

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/async/AsyncManager.java (hunk 2 above)
Comment (TylerS1066) [reply modules/Movecraft/src/main/java/net/countercraft/movecraft/async/AsyncManager.java:671]: After reading further, you should be able to remove the `instanceof` check here.

**Review by TylerS1066 — CHANGES_REQUESTED** [review]
I've not yet read status sign or considered the large scale effects of this PR, but here's an initial review.

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/craft/BaseCraft.java (hunk 3 above)
Comment (drfiveminusmint) [reply modules/Movecraft/src/main/java/net/countercraft/movecraft/craft/BaseCraft.java]: This was a bit of debug code I left in, whoops!

**Review by TylerS1066 — APPROVED** [review]

**Pull request merged**